export/securedrop_export/disk/cli.py

```diff
@@ -1,429 +1,526 @@
 
         finally:
-            self.cleanup_drive_and_tmpdir(device, submission_tmpdir)
+            self.cleanup(device, archive_tmpdir, is_error)
 
-    def cleanup_drive_and_tmpdir(self, volume: MountedVolume, submission_tmpdir: str):
+    def cleanup(
+        self,
+        volume: MountedVolume,
+        archive_tmpdir: str,
+        is_error: bool = False,
+        should_close_volume: bool = True,
+    ):
         """
         Post-export cleanup method. Unmount and lock drive and remove temporary
-        directory. Currently called at end of `write_data_to_device()` to ensure
-        device is always locked after export.
+        directory.
+
+        Raises ExportException if errors during cleanup are encountered.
 
-        Raise ExportException if errors during cleanup are encountered.
+        Method is called whether or not export succeeds; if `is_error` is True,
+        will report export error status on error (insted of cleanup status).
         """
+        error_status = Status.ERROR_EXPORT if is_error else Status.ERROR_EXPORT_CLEANUP
+
         logger.debug("Syncing filesystems")
         try:
             subprocess.check_call(["sync"])
-            umounted = self._unmount_volume(volume)
-            if umounted:
-                self._close_luks_volume(umounted)
-            self._remove_temp_directory(submission_tmpdir)
+            self._remove_temp_directory(archive_tmpdir)
+
+            # Future configurable option
+            if should_close_volume:
+                self._close_volume(volume)
 
         except subprocess.CalledProcessError as ex:
             logger.error("Error syncing filesystem")
-            raise ExportException(sdstatus=Status.ERROR_EXPORT_CLEANUP) from ex
+            raise ExportException(sdstatus=error_status) from ex
 
-    def _unmount_volume(self, volume: MountedVolume) -> Volume:
+    def _close_volume(self, mv: MountedVolume) -> Volume:
         """
-        Helper. Unmount volume
+        Unmount and close volume.
         """
-        if os.path.exists(volume.mountpoint):
-            logger.debug(f"Unmounting drive from {volume.mountpoint}")
-            try:
-                subprocess.check_call(["sudo", "umount", volume.mountpoint])
-
-            except subprocess.CalledProcessError as ex:
-                logger.error("Error unmounting device")
-                raise ExportException(sdstatus=Status.DEVICE_ERROR) from ex
-        else:
-            logger.info("Mountpoint does not exist; volume was already unmounted")
+        logger.debug(f"Unmounting drive {mv.unlocked_name} from {mv.mountpoint}")
+        try:
+            subprocess.check_call(
+                [
+                    "udisksctl",
+                    "unmount",
+                    "--block-device",
+                    f"{mv.unlocked_name}",
+                ],
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL,
```

Comment:
Since this fixes a stumper of a bug, let's document it:

```suggestion
                # Redirect stdout/stderr to avoid breaking the qrexec pipe:
                stdout=subprocess.DEVNULL,
                stderr=subprocess.DEVNULL,
```

Reply:
Yup, good call, done in https://github.com/freedomofpress/securedrop-client/pull/1777/commits/dc32ba2f8cdc4412020eff3679349e11596e037a